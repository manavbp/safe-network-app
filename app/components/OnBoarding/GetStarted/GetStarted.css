.Base {
    width: 100%;
    height: 100%;
    min-height: 600px;
    background: linear-gradient(156.98deg, #fa541c -164.31%, #454545 101.04%);
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
}

.LogoBase {
    display: flex;
    width: 80px;
    height: 80px;
    background-color: #fa541c;
    border-radius: 50%;
    justify-content: center;
    box-shadow: 0 0 100px #fa541c, 0 0 50px #fa541c,
        0 0 23px rgba(0, 0, 0, 0.25);
}

.Container {
    text-align: center;
    margin-top: 60px;
    color: #fff;
    padding: 0 40px;
}

.Title {
    margin-bottom: 20px;
}

.GetStartedButton {
    text-transform: uppercase;
    height: 36px;
    margin-top: 30px;
    background-color: #ff5722;
    color: #fff;
}
