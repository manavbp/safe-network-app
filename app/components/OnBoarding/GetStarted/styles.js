import deepOrange from '@material-ui/core/colors/deepOrange';
import Paper from '@material-ui/core/Paper';
import Box from '@material-ui/core/Box';
import Typography from '@material-ui/core/Typography';
import Fab from '@material-ui/core/Fab';
import { styled } from '@material-ui/core/styles';

export const Base = styled( Paper )( {
    width: '100%',
    height: '100%',
    minHeight: '600px',
    background: 'linear-gradient(156.98deg, #FA541C -164.31%, #454545 101.04%)',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column'
} );

export const LogoBase = styled( Box )( {
    display: 'flex',
    width: '80px',
    height: '80px',
    backgroundColor: '#FA541C',
    borderRadius: '50%',
    justifyContent: 'center',
    boxShadow:
        '0px 0px 100px #FA541C, 0px 0px 50px #FA541C, 0px 0px 23px rgba(0, 0, 0, 0.25)'
} );

export const Container = styled( Box )( {
    textAlign: 'center',
    marginTop: '60px',
    color: '#fff',
    padding: '0 40px'
} );

export const Title = styled( Typography )( {
    marginBottom: '20px'
} );

export const GetStartedButton = styled( Fab )( {
    textTransform: 'uppercase',
    // height: '36px',
    marginTop: '30px',
    backgroundColor: deepOrange[500],
    color: '#fff'
} );
