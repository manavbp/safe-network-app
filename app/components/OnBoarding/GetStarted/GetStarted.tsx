import React from 'react';
import Typography from '@material-ui/core/Typography';
import Paper from '@material-ui/core/Paper';
import Box from '@material-ui/core/Box';
import Fab from '@material-ui/core/Fab';
// @ts-ignore
import { INTRO } from '$App/constants/routes.json';

// @ts-ignore
import Logo from '$App/assets/images/logo.svg';
import styles from './GetStarted.css';

interface Props {
    onClickGetStarted: Function;
    history?: History;
}

export const GetStarted = ( properties ) => {
    const { onClickGetStarted, history } = properties;
    return (
        <Paper className={styles.Base} aria-label="GetStarted">
            <Box className={styles.LogoBase}>
                <img src={Logo} alt="Launchpad logo" />
            </Box>
            <Box className={styles.Container}>
                <Typography className={styles.Title} variant="h5">
                    SAFE Launchpad
                </Typography>
                <Typography>
                    All the apps you need to try the SAFE Network
                </Typography>
                <Fab
                    className={styles.GetStartedButton}
                    variant="extended"
                    size="medium"
                    aria-label="GetStarted"
                    onClick={() => {
                        onClickGetStarted();
                        // @ts-ignore
                        history.push( INTRO );
                    }}
                >
                    Get Started
                </Fab>
            </Box>
        </Paper>
    );
};
