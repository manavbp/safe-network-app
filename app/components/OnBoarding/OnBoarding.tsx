import React, { Component } from 'react';
import { History } from 'history';
import Grid from '@material-ui/core/Grid';
import { Switch, Route } from 'react-router';
import { styled } from '@material-ui/core/styles';
import {
    UserPreferences,
    AppPreferences,
    Preferences,
    LaunchpadState
} from '$Definitions/application.d';

import { Stepper } from './Stepper/Stepper';
import { GetStarted } from './GetStarted';
import { Intro } from './Intro';
import { BasicSettings } from './BasicSettings/BasicSettings';
import {
    HOME,
    INTRO,
    BASIC_SETTINGS,
    ON_BOARDING
} from '$Constants/routes.json';

interface Props {
    setAppPreferences: Function;
    aliasShouldOnboard: Function;
    triggerSetAsTrayWindow: Function;
    userPreferences: UserPreferences;
    appPreferences: AppPreferences;
    setUserPreferences: Function;
    getUserPreferences: Function;
    storePreferences: Function;
    isTrayWindow: boolean;
    setOnboardCompleted: Function;
    autoLaunch: Function;
    history?: History;
}

const Base = styled( Grid )( {
    position: 'absolute',
    top: 0,
    left: 0,
    width: '100%',
    height: '100%'
} );

export class OnBoarding extends React.Component<Props> {
    state = {
        currentPosition: 0
    };

    totalSteps = 3;

    componentWillMount() {
        const { appPreferences, history, getUserPreferences } = this.props;

        if ( !appPreferences.shouldOnboard ) {
            history.push( HOME );
        }
        getUserPreferences();
    }

    completed = () => {
        const {
            storePreferences,
            userPreferences,
            autoLaunch,
            setAppPreferences
        } = this.props;

        const appPreferences: AppPreferences = {
            shouldOnboard: false
        };

        const newUserPreferences: UserPreferences = {
            ...userPreferences
        };

        const preferences: Preferences = {
            userPreferences: { ...newUserPreferences },
            appPreferences
        };

        storePreferences( preferences );
        setAppPreferences( { shouldOnboard: false } );

        if ( userPreferences.launchOnStart ) {
            autoLaunch( true );
        }
    };

    onNext = () => {
        const { currentPosition } = this.state;
        const position = currentPosition;

        if ( currentPosition < this.totalSteps - 1 ) {
            this.setState( {
                currentPosition: currentPosition + 1
            } );
        } else {
            this.completed();
        }
    };

    onBack = () => {
        const { currentPosition } = this.state;
        const position = currentPosition;

        if ( currentPosition > 0 ) {
            this.setState( {
                currentPosition: currentPosition - 1
            } );
        }
    };

    render() {
        const { currentPosition } = this.state;
        const {
            userPreferences,
            setUserPreferences,
            getUserPreferences,
            storePreferences,
            isTrayWindow,
            triggerSetAsTrayWindow,
            autoLaunch,
            history
        } = this.props;

        const isGetStarted = currentPosition === 0;

        return (
            <Base container>
                <Grid item xs={12}>
                    <Switch>
                        <Route
                            exact
                            path={ON_BOARDING}
                            component={() => (
                                <GetStarted
                                    onClickGetStarted={this.onNext}
                                    // @ts-ignore
                                    history={history}
                                />
                            )}
                        />
                        <Route path={INTRO} component={() => <Intro />} />
                        <Route
                            path={BASIC_SETTINGS}
                            component={() => (
                                <BasicSettings
                                    userPreferences={userPreferences}
                                    setUserPreferences={setUserPreferences}
                                    triggerSetAsTrayWindow={
                                        triggerSetAsTrayWindow
                                    }
                                    autoLaunch={autoLaunch}
                                    isTrayWindow={isTrayWindow}
                                />
                            )}
                        />
                    </Switch>
                </Grid>
                <Stepper
                    theme={isGetStarted ? 'white' : 'default'}
                    showButtons={!isGetStarted}
                    noElevation={isGetStarted}
                    onNext={this.onNext}
                    onBack={this.onBack}
                    steps={this.totalSteps}
                    activeStep={this.state.currentPosition}
                    // @ts-ignore
                    history={history}
                />
            </Base>
        );
    }
}
