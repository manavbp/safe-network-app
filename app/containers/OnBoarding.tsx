import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { OnBoarding } from '$Components/OnBoarding/OnBoarding';

import {
    setUserPreferences,
    setAppPreferences,
    getUserPreferences
} from '$Actions/launchpad_actions';
import {
    autoLaunch,
    triggerSetAsTrayWindow
} from '$Actions/alias/launchpad_actions';

const mapStateToProperties = ( state ) => {
    return {
        userPreferences: state.launchpad.userPreferences,
        appPreferences: state.launchpad.appPreferences,
        isTrayWindow: state.launchpad.isTrayWindow
    };
};

const mapDispatchToProperties = ( dispatch ) => {
    const actions = {
        setAppPreferences,
        setUserPreferences,
        getUserPreferences,
        autoLaunch,
        triggerSetAsTrayWindow
    };
    return bindActionCreators( actions, dispatch );
};

export const OnBoardingPage: React.ComponentClass = connect(
    mapStateToProperties,
    mapDispatchToProperties
)( OnBoarding );
