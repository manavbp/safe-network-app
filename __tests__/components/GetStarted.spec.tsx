import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';

import { GetStarted } from '$App/components/OnBoarding/GetStarted';

const shallow = createShallow();

const shallowSetup = ( propOverrides? ) => {
    const props = {
        history: {
            push: jest.fn()
        },
        ...Object.assign(
            {
                onClickGetStarted: jest.fn()
            },
            propOverrides
        )
    };

    const wrapper = shallow( <GetStarted {...props} /> );

    return {
        props,
        wrapper
    };
};

const getContainer = ( wrapper ) => wrapper.childAt( 1 );

describe( 'Get Started', () => {
    it( 'render', () => {
        const { wrapper } = shallowSetup();
        expect( wrapper ).toMatchSnapshot();
    } );

    it( 'title should be right', () => {
        const { wrapper } = shallowSetup();
        expect(
            getContainer( wrapper )
                .childAt( 0 )
                .text()
        ).toEqual( 'SAFE Launchpad' );
        expect(
            getContainer( wrapper )
                .childAt( 0 )
                .prop( 'variant' )
        ).toEqual( 'h5' );
    } );

    it( 'description should be right', () => {
        const { wrapper } = shallowSetup();
        expect(
            getContainer( wrapper )
                .childAt( 1 )
                .text()
        ).toEqual( 'All the apps you need to try the SAFE Network' );
    } );

    it( 'clicking Get Started button works', () => {
        const { wrapper, props } = shallowSetup();
        getContainer( wrapper )
            .childAt( 2 )
            .props()
            .onClick();
        expect( props.onClickGetStarted ).toHaveBeenCalled();
    } );
} );
