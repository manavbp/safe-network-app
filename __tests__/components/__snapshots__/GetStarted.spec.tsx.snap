// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get Started render 1`] = `ShallowWrapper {}`;
